### PR TITLE
Titania port

### PR DESCRIPTION
**Pull request by msavolainen-gofore**

### CODE REVIEW

File: service/src/main/kotlin/fi/espoo/evaka/titania/Titania.kt [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+
+enum class TitaniaPayrollItemUnit {
+    /** Minuutit */
+    MINUTE,
+
+    /** Kappaleet */
+    QUANTITY
+}
+
+data class UpdateWorkingTimeEventsRequest(
```
Comment (Wnt) [inline service/src/main/kotlin/fi/espoo/evaka/titania/Titania.kt:125]: Add `@ExcludeCodeGen` annotation to all route payload / return types to not have them included in `frontend/src/lib-common/generated/api-types/`
```suggestion
@ExcludeCodeGen
data class UpdateWorkingTimeEventsRequest(
```

**Review by Gekkio — COMMENTED** [review]
One general comment: I think `internal` never useful, and things should generally be just public (no visibility keywords) or `private` if you really want to hide something.

File: service/src/integrationTest/kotlin/fi/espoo/evaka/titania/TitaniaControllerTest.kt [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+import org.mockito.kotlin.verifyNoInteractions
+import org.mockito.kotlin.whenever
+import org.skyscreamer.jsonassert.JSONAssert.assertEquals
+import org.skyscreamer.jsonassert.JSONCompareMode
+import org.springframework.boot.test.mock.mockito.MockBean
+import org.springframework.core.io.ClassPathResource
+import org.springframework.http.HttpStatus
+import org.springframework.util.StreamUtils
+
+internal class TitaniaControllerTest : FullApplicationTest(resetDbBeforeEach = true) {
```
[the remark this answers is not in the document]
Comment (vesahonkasaari) [reply service/src/integrationTest/kotlin/fi/espoo/evaka/titania/TitaniaControllerTest.kt:19]: Original TitaniaControllerTest was entirely replaced with tests which call the HTTP endpoints to verify the status code handling

File: service/src/main/kotlin/fi/espoo/evaka/EvakaEnv.kt [hunk 3]
```diff
@@ -98,7 +99,8 @@ data class EvakaEnv(
                 feeDecisionGeneratorV2Enabled =
                     env.lookup("evaka.fee_decision.generator.v2_enabled") ?: false,
                 voucherValueDecisionGeneratorV2Enabled =
-                    env.lookup("evaka.voucher_value_decision.generator.v2_enabled") ?: false
+                    env.lookup("evaka.voucher_value_decision.generator.v2_enabled") ?: false,
+                titaniaEnabled = env.lookup("evaka.titania_enabled") ?: false
```
Comment (Gekkio) [inline service/src/main/kotlin/fi/espoo/evaka/EvakaEnv.kt:103]: This should probably be removed since it doesn't seem to be now used anywhere (?)

Comment (vesahonkasaari) [reply service/src/main/kotlin/fi/espoo/evaka/EvakaEnv.kt]: Good catch, fixed.

**Review by Gekkio — APPROVED** [review]

**Pull request merged**